### PR TITLE
chore: adds queue uri json generation

### PR DESCRIPTION
**Pull request by xar-tol**
Adds generation of COPILOT_QUEUE_URIS with worker services.

Addresses #2550

### CODE REVIEW

File: templates/workloads/partials/cf/envvars.yml [hunk 1]
```diff
@@ -23,6 +23,17 @@ Environment:
...
   Value: '{{jsonSNSTopics .Publish.Topics}}'
 {{- end}}{{- end}}
+{{- if .Subscribe}}{{- if .Subscribe.Topics}}
+- Name: COPILOT_QUEUE_URIS
+  Value: !Sub
+    - '{{jsonQueueURIs .Subscribe.Topics}}'
+    - mainURL: !Ref EventsQueue
+    {{- range $topic := .Subscribe.Topics}}
+    {{- if $topic.Queue}}
+      {{$topic.Service}}-{{$topic.Name}}URL: !Ref {{logicalIDSafe $topic.Service}}{{logicalIDSafe $topic.Name}}EventsQueue
```
Comment (Lou1415926) [inline templates/workloads/partials/cf/envvars.yml]: Does it render properly when `$topic.Service` or `$topic.Name` is `nil`? I don't recall 😛

**Review by Lou1415926 — COMMENTED** [review]
Neat! This would be so helpful to users!

File: templates/workloads/partials/cf/envvars.yml (hunk 1 above)
Comment (xar-tol) [reply templates/workloads/partials/cf/envvars.yml]: If nil it should skip them as they need a valid name and service to be subscribed to. I added more ifs.

File: templates/workloads/partials/cf/envvars.yml [hunk 2]
```diff
@@ -28,11 +28,9 @@ Environment:
   Value: !Sub
     - '{{jsonQueueURIs .Subscribe.Topics}}'
     - mainURL: !Ref EventsQueue
-    {{- range $topic := .Subscribe.Topics}}
-    {{- if $topic.Queue}}
-      {{$topic.Service}}-{{$topic.Name}}URL: !Ref {{logicalIDSafe $topic.Service}}{{logicalIDSafe $topic.Name}}EventsQueue
-    {{- end}}
-    {{- end}}
+    {{- range $topic := .Subscribe.Topics}}{{- if $topic.Queue}}{{- if $topic.Service}}{{- if $topic.Name}}
```
Comment (Lou1415926) [inline templates/workloads/partials/cf/envvars.yml]: We can make use of `and` function: https://pkg.go.dev/text/template#hdr-Functions
```suggestion
    {{- range $topic := .Subscribe.Topics}}{{- if and $topic.Queue $topic.Service $topic.Name}}
```

File: internal/pkg/template/template_functions.go [hunk 3]
```diff
@@ -154,6 +154,36 @@ func generateSNSJSON(topics []*Topic) string {
 	return string(out)
 }
 
+// generateQueueURIJSON turns a list of Topic Subscription objects into a JSON string of their corresponding queues:
+// `{"EventsQueue": "${mainURL}", "svc-topicEventsQueue": "${svc-topicURL}"}`
```
Comment (Lou1415926) [inline internal/pkg/template/template_functions.go]: We should update the comment as well!

File: internal/pkg/template/template_functions.go [hunk 4]
```diff
@@ -154,6 +154,36 @@ func generateSNSJSON(topics []*Topic) string {
...
+// This function must be called on an array of correctly constructed Topic objects.
+func generateQueueURIJSON(ts []*TopicSubscription) string {
+	if ts == nil {
+		return ""
+	}
+	urlMap := make(map[string]string)
+	urlMap["EventsQueue"] = "${mainURL}"
```
Comment (Lou1415926) [inline internal/pkg/template/template_functions.go]: We should probably keep it consistent with the other json injection such that the key is `snakeCaseCaseSnake` - sorry for not seeing this in my previous review!

**Review by Lou1415926 — COMMENTED** [review]
Looks good! Just two more nits 🙇🏼‍♀️

File: internal/pkg/template/template_functions.go (hunk 4 above)
Comment (Lou1415926) [reply internal/pkg/template/template_functions.go]: So the json string would be 
`{"eventsQueue": "${mainURL}", "svcTopicEventsQueue": "${svctopicURL}"}`

**Review by bvtujo — APPROVED** [review]

File: templates/workloads/partials/cf/envvars.yml (hunk 2 above)
Comment (bvtujo) [reply templates/workloads/partials/cf/envvars.yml]: does this work with more than two arguments?

Comment (Lou1415926) [reply templates/workloads/partials/cf/envvars.yml]: yess it does

**Review by Lou1415926 — APPROVED** [review]

**Pull request merged**